### PR TITLE
Refactor TCP Metrics task for telemetry v2

### PR DESCRIPTION
**Pull request by gargnupur**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by gargnupur** [comment]
@douglas-reid , @mandarjog , @kyessenov : Should we add a flow diagram of how TCP metrics is using ALPN and prefix based approach now?

**Comment by ericvn** [comment]
There are also some formatting errors in the _lint_ test.

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 1]
```diff
@@ -107,6 +90,30 @@ will generate and collect automatically.
         Created config virtual-service/default/ratings at revision 3004
         {{< /text >}}
 
+    1.  Cleanup mongo proxy as the terminal filter:
```
Comment (mandarjog) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Please add a comment at the top about this being temporary. It is embedded in the config, but should be at the top as well.

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 2]
```diff
@@ -125,54 +132,23 @@ will generate and collect automatically.
...
-    View values for the new metric in the Prometheus browser window.  Select **Graph**.
-    Enter the `istio_mongo_received_bytes` metric and select **Execute**.
+    View values for the TCP metrics in the Prometheus browser window.  Select **Graph**.
+    Enter the `istio_tcp_connections_opened_total` metric and select **Execute**.
     The table displayed in the
     **Console** tab includes entries similar to:
 
     {{< text plain >}}
-    istio_mongo_received_bytes{destination_version="v1",instance="172.17.0.18:42422",job="istio-mesh",source_service="ratings-v2",source_version="v2"}
+    istio_tcp_connections_opened_total{destination_version="v1",instance="172.17.0.18:42422",job="istio-mesh",source_service="ratings-v2",source_version="v2"}
```
Comment (mandarjog) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: We should have both opened_total and received_bytes as example.

**Comment by mandarjog** [comment]
I am wondering If we should add the nuance about calling out of mesh service, and not having full client side telemetry around it. @kyessenov @douglas-reid

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 1 above)
Comment (kyessenov) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: I don't think this should surface in operator-visible docs. This is fine for the developer docs, but it is extremely ugly.

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Removed this config..

**Comment by douglas-reid** [comment]
> I am wondering If we should add the nuance about calling out of mesh service, and not having full client side telemetry around it. @kyessenov @douglas-reid

I don't think we need to put that into the specific in-mesh TCP task. That should live somewhere else, I think.

**Comment by douglas-reid** [comment]
> @douglas-reid , @mandarjog , @kyessenov : Should we add a flow diagram of how TCP metrics is using ALPN and prefix based approach now?

I would like to see one, if we have it. It would match the old TCP metrics task nicely.

**Comment by gargnupur** [comment]
> We need to move the old TCP content some first before we start deleting it from this location, right? Otherwise we'll lose the info on the task for v1.

@douglas-reid : was planning to rebase and submit this after your pr https://github.com/istio/istio.io/pull/6533/files..

**Comment by gargnupur** [comment]
Add WIP label as need to change based on docs refactoring after https://github.com/istio/istio.io/pull/6533

**Comment by gargnupur** [comment]
@douglas-reid : Can you please take a look? I have rebased it after your changes..

**Comment by gargnupur** [comment]
@sdake , @adammil2000 , @frankbu : can one of you please review this too?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
+---
+title: Collecting Metrics for TCP services
+description: This task shows you how to configure Istio to collect metrics for TCP services.
+weight: 20
+keywords: [telemetry,metrics,tcp]
+aliases:
+    - /docs/tasks/telemetry/tcp-metrics
```
Comment (ericvn) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: If this is a totally new file and there aren't any files being deleted (just looking at this PR), is there a need for the aliases?

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: This is a new file for old tcp-metrics page. So, keep the aliases?

**Comment by gargnupur** [comment]
@douglas-reid : does https://preliminary.istio.io/docs/tasks/observability/metrics/querying-metrics/ cover HTTP?

**Comment by gargnupur** [comment]
@ericvn : Can you please review this again?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+In this task, you used Istio configuration to
+automatically generate and report metrics for all traffic to a TCP service
+within the mesh.
+
+### TCP attributes
+
+Several TCP-specific attributes enable TCP policy and control within Istio.
+These attributes are generated by Envoy Proxies and obtained from Istio using Envoy's Node Metadata.
+Peer Envoys are forwarded Node Metadata based on ALPN based tunneling with a prefixed based protocol.
```
Comment (adammil2000) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Is there a way to rewrite this sentence to be more clear? It's hard to parse. Something more like "X forwards Node Metadata to Peer Envoys using ALPN based tunneling and a prefixed protocol" or similar?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+We define a new protocol `istio-peer-exchange`, that is advertised and prioritized by the client and the server sidecars
+in the mesh. ALPN negotiation would resolve the protocol to `istio-peer-exchange` for connections between Istio enabled
```
Comment (adammil2000) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]:
```suggestion
in the mesh. ALPN negotiation resolves the protocol to `istio-peer-exchange` for connections between Istio enabled
```

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+proxies, but not between an Istio enabled proxy and any other proxy.
+This protocol extends TCP as follows:
+
+1.  TCP client, as a first sequence of bytes, sends a magic byte string and a length prefixed payload.
+1.  TCP server, as a first sequence of bytes, sends a magic byte sequence and a length prefix payload. These payloads
+ are protobuf encoded serialized metadata.
```
Comment (adammil2000) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Is the word magic used here the real technical term?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+1.  Client and server can write simultaneously and out of order. The extension filter in Envoy should hold the further
```
Comment (adammil2000) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: "Envoy should hold", do you mean "Envoy holds"? The word "should" in this location confuses the meaning.

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 4 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: How about?
Envoy forwards Node Metadata to Peer Envoys using ALPN based tunneling and a prefixed protocol

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 6 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: yeah :)

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 7 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: "The extension filter in Envoy then does the further" -> how about this?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 3 above)
Comment (ericvn) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Yes. That way old URL still comes here.

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        {{< text bash >}}
+        $ kubectl apply -f @samples/bookinfo/networking/virtual-service-ratings-db.yaml@
+        Created config virtual-service/default/reviews at revision 3003
+        Created config virtual-service/default/ratings at revision 3004
+        {{< /text >}}
+
+1.  Send traffic to the sample application.
+
+    For the Bookinfo sample, visit `http://$GATEWAY_URL/productpage` in your web
+    browser or use the following command:
```
Comment (ericvn) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: I'm not sure how far down the `Deploy Bookinfo` the user would have gotten in the prior step. GATEWAY_URL isn't set until the `Confirm the app is accessible from outside the cluster` step. The step after that is applying the `destination-rule-all` yaml which you do call out here. Maybe @adammil2000 or @frankbu have more experience on what it means to say `Deploy Bookinfo` and how far the user may have gone.

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 4 above)
Comment (ericvn) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Agree the original sentence is confusion. This sentence is better. Should it be `prefix based` or `prefixed based` protocol (no _based_ in the later sentence)?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 6 above)
Comment (ericvn) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Should this sentence have `length prefixed` like the others?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 8 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: We did similar thing for previous tcp metrics task too and don't think there was a problem... https://preliminary.istio.io/docs/tasks/observability/mixer/metrics/tcp-metrics/

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 4 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Thanks, corrected and yes it should be prefix based

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 6 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: changed to `prefix` everywhere

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Changed to `prefix based protocol` and `length prefixed payload`.. please review

**Comment by gargnupur** [comment]
@adammil2000 , @ericvn : it's ready for review again

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 8 above)
Comment (adammil2000) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Eric is correct, you're missing the steps where a value actually gets assigned to GATEWAY_URL to use it here. Check out the steps we used here to make that happen correctly: https://preliminary.istio.io/docs/setup/getting-started/#determining-the-ingress-ip-and-ports

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md]: Thanks.. Fixed

**Comment by gargnupur** [comment]
@adammil2000 , @ericvn : ptal.. I think this page is needed for 1.5 release...
